x/posts/keeper/keeper.go

```diff
@@ -40,75 +46,79 @@ func NewKeeper(cdc *codec.Codec, storeKey sdk.StoreKey, paramSpace params.Subspa
 // It assumes that the given post has already been validated.
 // If another post has the same ID of the given post, the old post will be overridden
 func (k Keeper) SavePost(ctx sdk.Context, post types.Post) {
-	store := ctx.KVStore(k.StoreKey)
+	store := ctx.KVStore(k.storeKey)
 
 	// Save the post
-	store.Set(types.PostStoreKey(post.PostID), k.Cdc.MustMarshalBinaryBare(&post))
+	store.Set(types.PostStoreKey(post.PostID), k.cdc.MustMarshalBinaryBare(&post))
 
 	// Check if the postID got an associated post, if not, increment the number of posts
 	if !store.Has(types.PostIndexedIDStoreKey(post.PostID)) {
 		// Retrieve the total number of posts, if null it will be equal to 0
-		numberOfPosts := sdk.ZeroInt()
+		postIndex := types.PostIndex{Value: 0}
 		if store.Has(types.PostTotalNumberPrefix) {
-			k.Cdc.MustUnmarshalBinaryBare(store.Get(types.PostTotalNumberPrefix), &numberOfPosts)
+			k.cdc.MustUnmarshalBinaryBare(store.Get(types.PostTotalNumberPrefix), &postIndex)
 		}
 
-		numberOfPosts = numberOfPosts.Add(sdk.NewInt(1))
+		postIndex = types.PostIndex{Value: postIndex.Value + 1}
 
 		// Save the new incremental ID of the post and update the total number of posts
-		store.Set(types.PostIndexedIDStoreKey(post.PostID), k.Cdc.MustMarshalBinaryBare(&numberOfPosts))
-		store.Set(types.PostTotalNumberPrefix, k.Cdc.MustMarshalBinaryBare(&numberOfPosts))
+		store.Set(types.PostIndexedIDStoreKey(post.PostID), k.cdc.MustMarshalBinaryBare(&postIndex))
+		store.Set(types.PostTotalNumberPrefix, k.cdc.MustMarshalBinaryBare(&postIndex))
 	}
 
 	// Save the comments to the parent post, if it is valid
-	if post.ParentID.Valid() {
+	if types.IsValidPostID(post.ParentID) {
 		parentCommentsKey := types.PostCommentsStoreKey(post.ParentID)
 
-		var commentsIDs types.PostIDs
-		k.Cdc.MustUnmarshalBinaryBare(store.Get(parentCommentsKey), &commentsIDs)
+		var commentsIDs types.CommentIDs
+		k.cdc.MustUnmarshalBinaryBare(store.Get(parentCommentsKey), &commentsIDs)
 		if editedIDs, appended := commentsIDs.AppendIfMissing(post.PostID); appended {
-			store.Set(parentCommentsKey, k.Cdc.MustMarshalBinaryBare(&editedIDs))
+			store.Set(parentCommentsKey, k.cdc.MustMarshalBinaryBare(&editedIDs))
 		}
 	}
 }
 
+// DoesPostExist returns true if the post with the given id exists inside the store.
+func (k Keeper) DoesPostExist(ctx sdk.Context, id string) bool {
+	store := ctx.KVStore(k.storeKey)
+	return store.Has(types.PostStoreKey(id))
+}
+
 // GetPost returns the post having the given id inside the current context.
 // If no post having the given id can be found inside the current context, false will be returned.
-func (k Keeper) GetPost(ctx sdk.Context, id types.PostID) (post types.Post, found bool) {
-	store := ctx.KVStore(k.StoreKey)
-
-	key := types.PostStoreKey(id)
-	if !store.Has(key) {
+func (k Keeper) GetPost(ctx sdk.Context, id string) (post types.Post, found bool) {
+	if !k.DoesPostExist(ctx, id) {
```

Comment:
I think we could avoid this and check this later by doing:
```suggestion
	store := ctx.KVStore(k.storeKey)
	bz := store.Get(types.PostStoreKey(id))
	if bz == nil {
	        return types.Post{}, false
	}
	k.cdc.MustUnmarshalBinaryBare(bz, &post)
```